src/SFA.DAS.EmployerIncentives.Data/ApprenticeshipIncentives/ApprenticeshipIncentiveDataRepository.cs

```diff
@@ -151,6 +129,34 @@ public async Task Delete(ApprenticeshipIncentiveModel apprenticeshipIncentive)
             }
         }
 
+        private async Task<ApprenticeshipIncentiveModel> Get(Expression<Func<ApprenticeshipIncentive, bool>> predicate)
+        {
+            var apprenticeshipIncentive = await _dbContext.ApprenticeshipIncentives.FirstOrDefaultAsync(predicate);
+
+            if (apprenticeshipIncentive != null)
+            {
+                await PopulateApprenticeshipIncentive(apprenticeshipIncentive);
+                return apprenticeshipIncentive?.Map(_dbContext.CollectionPeriods.AsEnumerable());
+            }
+
+            return null;
+        }
+
+        private async Task PopulateApprenticeshipIncentive(ApprenticeshipIncentive apprenticeshipIncentive)
+        {
+            apprenticeshipIncentive.PendingPayments = await _dbContext.PendingPayments.Where(x => x.ApprenticeshipIncentiveId == apprenticeshipIncentive.Id).ToListAsync();
+            foreach (var pendingPayment in apprenticeshipIncentive.PendingPayments)
+            {
+                pendingPayment.ValidationResults = await _dbContext.PendingPaymentValidationResults.Where(x => x.PendingPaymentId == pendingPayment.Id).ToListAsync();
```

Comment:
There's no need for this for each loop. Not all records need to be singularly fetched.   The PendingPaymentValidationResults can come back with the PendingPayments so just use the Include() in this case.

Reply:
Updated